lib/bolt/inventory/group.rb

```diff
@@ -9,30 +9,47 @@ class Group
 
       def initialize(data)
         @logger = Logging.logger[self]
-        @name = data['name']
-        @nodes = {}
 
-        data['nodes']&.each do |n|
-          n = { 'name' => n } if n.is_a? String
-          if @nodes.include? n['name']
-            @logger.warn("Ignoring duplicate node in #{@name}: #{n}")
+        if data.key?('name')
```

Comment:
Do we need to make sure this is a hash?

Reply:
I'm not sure! Without full schema validation, there will definitely be some scenarios that aren't covered. I was trying to think of cases where users might inadvertently enter the wrong thing (like comma-separated string for `nodes` instead of an array). So I was unclear whether the `groups` list could reasonably be misinterpreted.

I can certainly add a check for it though.